cluster/router/v3router/judger/attachment_match_judger.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package judger
+
+import (
+	"github.com/apache/dubbo-go/config"
+	"github.com/apache/dubbo-go/protocol"
+)
+
+type AttachmentMatchJudger struct {
+	config.DubboAttachmentMatch
+}
+
+// nolint
+func (amj *AttachmentMatchJudger) Judge(invocation protocol.Invocation) bool {
+	invAttaMap := invocation.Attachments()
+	if amj.EagleeyeContext != nil {
+		for k, v := range amj.EagleeyeContext {
+			invAttaValue, ok := invAttaMap[k]
+			if !ok {
+				if v.Empty == "" {
+					return false
+				}
+			}
+			// exist this key
+			str, ok := invAttaValue.(string)
+			if !ok {
+				return false
+			}
+			strJudger := NewStringMatchJudger(v)
+			if !strJudger.Judge(str) {
+				return false
+			}
+		}
+	}
+
+	if amj.DubboContext != nil {
+		for k, v := range amj.DubboContext {
+			invAttaValue, ok := invAttaMap[k]
+			if !ok {
+				if v.Empty == "" {
+					return false
+				}
+			}
+			// exist this key
+			str, ok := invAttaValue.(string)
```

Comment:
`invAttaValue` is nil when `ok==false`, is that correct?

Reply:
Yes I fixed it.